### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by dolgopolovwork**
Just README.md

### CODE REVIEW

**Review by Mingela — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Reverse chain adapter
+## Service overview 
+Reverse chain adapter is a service that is used for handling crucial financial operations in Iroha Hyperledger.
```
Comment (LiraLemur) [inline README.md]: Hyperledger Iroha

File: README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Reverse chain adapter
+## Service overview 
+Reverse chain adapter is a service that is used for handling crucial financial operations in Iroha Hyperledger. 
+It uses RabbitMQ as middleware between Iroha clients and Iroha peer. 
+Using the service, transactions are not sent directly to Iroha peer but stored in a dedicated persistent queue.
```
Comment (LiraLemur) [inline README.md]: With this service on, transactions...

File: README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# Reverse chain adapter
+## Service overview 
+Reverse chain adapter is a service that is used for handling crucial financial operations in Iroha Hyperledger. 
+It uses RabbitMQ as middleware between Iroha clients and Iroha peer. 
+Using the service, transactions are not sent directly to Iroha peer but stored in a dedicated persistent queue. 
+Once a transaction is saved in the queue, the service will read the incoming transaction and try to send it to Iroha peer.
```
Comment (LiraLemur) [inline README.md]: Once a transaction is saved in the queue, the service will read that transaction and try to send it to Iroha peer.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+# Reverse chain adapter
+## Service overview 
+Reverse chain adapter is a service that is used for handling crucial financial operations in Iroha Hyperledger. 
+It uses RabbitMQ as middleware between Iroha clients and Iroha peer. 
+Using the service, transactions are not sent directly to Iroha peer but stored in a dedicated persistent queue. 
+Once a transaction is saved in the queue, the service will read the incoming transaction and try to send it to Iroha peer. 
+The service will send the transaction back to the queue if Iroha peer is not online, forming a loop. 
+The service will try to send every incoming transaction to Iroha peer unless it's not received no matter how many `send` attempts it might take.
```
Comment (LiraLemur) [inline README.md:8]: until it's received?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
+# Reverse chain adapter
+## Service overview 
+Reverse chain adapter is a service that is used for handling crucial financial operations in Iroha Hyperledger. 
+It uses RabbitMQ as middleware between Iroha clients and Iroha peer. 
+Using the service, transactions are not sent directly to Iroha peer but stored in a dedicated persistent queue. 
+Once a transaction is saved in the queue, the service will read the incoming transaction and try to send it to Iroha peer. 
+The service will send the transaction back to the queue if Iroha peer is not online, forming a loop. 
+The service will try to send every incoming transaction to Iroha peer unless it's not received no matter how many `send` attempts it might take. 
+The service will read all the incoming transactions from the queue on start-up and try to handle them if the service was offline by the time transaction was sent to RabbitMQ.
```
Comment (LiraLemur) [inline README.md:9]: all incoming transactions; if the service was offline when transaction was sent...

File: README.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+## Configuration file overview
+Reverse chain adapter uses `reverse-chain-adapter.properties` as a default configuration file that is located in the `resources` folder inside the project. However, every configuration item could be changed through environmental variables. 
+
+- `reverse-chain-adapter.rmqHost` - RabbitMQ host
+- `reverse-chain-adapter.rmqPort` - RabbitMQ port
+- `reverse-chain-adapter.transactionQueueName` - name of queue tha will be used to persist incoming transactions. The values equals to `reverse_adapter_queue` by default.
```
Comment (LiraLemur) [inline README.md:15]: of the queue that will be used for saving the incoming transactions?

File: README.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+```
+ReverseChainAdapter(reverseChainAdapterConfig, irohaAPI).init()
+    .fold(
+        { println("Reverse chain adapter has been started") },
+        { ex ->
+            ex.printStackTrace()
+            System.exit(1)
+        })
+```
+In this case no health check endpoint for the adapter will be run, so setting `healthCheckPort` value won't make any effect.
```
Comment (LiraLemur) [inline README.md:50]: have any effect

File: README.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  
+## How to use
+`com.d3.reverse.client.ReliableIrohaConsumerImpl` is the class used as a client for the service. The class may be obtained via [Jitpack](https://jitpack.io/#d3ledger/reverse-chain-adapter):
+
+```groovy
+implementation "com.github.d3ledger.reverse-chain-adapter:reverse-chain-adapter-client:$reverse_chain_adapter_client_version"
+``` 
+The client just sends your transaction to RabbitMQ and subscribes to its status. The client will "re-subscribe" to the transaction's status if Iroha is not online. If you are not interested in the transaction's status simply set the `fireAndForget` value to `true`.
```
Comment (LiraLemur) [inline README.md:58]: simply sends or only sends;

**Review by LiraLemur — APPROVED** [review]

**Pull request merged**